lib/query.js

```diff
@@ -3259,6 +3268,10 @@ Query.prototype.findOneAndUpdate = function(filter, doc, options) {
     delete options.fields;
   }
 
+  if (options.overwrite) {
```

Comment:
Check if `options.overwrite != null`, we should also warn if `overwrite: false` is set.

Also, you should move this to `_findOneAndUpdate()`, because your current implementation won't catch `findOneAndUpdate(filter, update).setOptions({ overwrite: true })`.